test/e2e/e2e_suite_test.go

```diff
@@ -19,16 +19,278 @@ limitations under the License.
+				return 0, err
+			}
+			return deployment.Status.ReadyReplicas, nil
+		}, 5*time.Minute, 15*time.Second,
+	).ShouldNot(BeZero())
+}
+
+func getSession() client.ConfigProvider {
+	sess, err := session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	})
+	Expect(err).NotTo(HaveOccurred())
+	return sess
+}
+
+func getAccountID(prov client.ConfigProvider) string {
+	stsSvc := sts.NewService(awssts.New(prov))
+	accountID, err := stsSvc.AccountID()
+	Expect(err).NotTo(HaveOccurred())
+	return accountID
+}
+
+func createIAMRoles(prov client.ConfigProvider, accountID string) {
+	cfnSvc := cloudformation.NewService(cfn.New(prov))
+	Expect(
+		cfnSvc.ReconcileBootstrapStack(stackName, accountID, "aws"),
+	).To(Succeed())
+}
+
+func createKeyPair(prov client.ConfigProvider) {
+	ec2c := ec2.New(prov)
+	_, err := ec2c.CreateKeyPair(&ec2.CreateKeyPairInput{KeyName: aws.String(keyPairName)})
+	if code, _ := awserrors.Code(err); code != "InvalidKeyPair.Duplicate" {
+		Expect(err).NotTo(HaveOccurred())
+	}
+}
+
+func loadManagerImage(kindCluster kind.Cluster) {
+	if managerImage != nil && *managerImage != "" {
+		kindCluster.LoadImage(*managerImage)
+	}
+}
+
+func applyManifests(kindCluster kind.Cluster, manifests *string) {
+	Expect(manifests).ToNot(BeNil())
+	fmt.Fprintf(GinkgoWriter, "Applying manifests for %s\n", *manifests)
+	Expect(*manifests).ToNot(BeEmpty())
+	kindCluster.ApplyYAML(*manifests)
+}
+
+func deployCAPAComponents(kindCluster kind.Cluster) {
+	if capaComponents != nil && *capaComponents != "" {
+		applyManifests(kindCluster, capaComponents)
+		return
+	}
+
+	fmt.Fprintf(GinkgoWriter, "Generating CAPA manifests\n")
+
+	// Build the manifests using kustomize
+	capaManifests, err := exec.Command(*kustomizeBinary, "build", "../../config/default").Output()
+	if err != nil {
+		if exitError, ok := err.(*exec.ExitError); ok {
+			fmt.Fprintf(GinkgoWriter, "Error: %s\n", string(exitError.Stderr))
+		}
+	}
+	Expect(err).NotTo(HaveOccurred())
+
+	// envsubst the credentials
+	b64credentials, err := generateB64Credentials()
+	Expect(err).NotTo(HaveOccurred())
+	os.Setenv("AWS_B64ENCODED_CREDENTIALS", b64credentials)
+	manifestsContent := os.ExpandEnv(string(capaManifests))
+
+	// write out the manifests
+	manifestFile := path.Join(suiteTmpDir, "infrastructure-components.yaml")
+	Expect(ioutil.WriteFile(manifestFile, []byte(manifestsContent), 0644)).NotTo(HaveOccurred())
+
+	// apply generated manifests
+	applyManifests(kindCluster, &manifestFile)
+}
+
+const AWSCredentialsTemplate = `[default]
+aws_access_key_id = {{ .AccessKeyID }}
+aws_secret_access_key = {{ .SecretAccessKey }}
+region = {{ .Region }}
+{{if .SessionToken }}
+aws_session_token = {{ .SessionToken }}
+{{end}}
+`
+
+type awsCredential struct {
+	AccessKeyID     string
+	SecretAccessKey string
+	SessionToken    string
+	Region          string
+}
+
+func generateB64Credentials() (string, error) {
```

Comment:
Could we use `clusterawsadm` for this?

Reply:
We could, but I wanted to avoid taking a hard dependency on the binary.